storage/engine.go

```diff
@@ -318,13 +320,22 @@ func (e *Engine) DeleteBucketRangePredicate(ctx context.Context, orgID, bucketID
 	return e.tsdbStore.DeleteSeriesWithPredicate(bucketID.String(), min, max, pred)
 }
 
+// LockKVStore locks the KV store as well as the engine in preparation for doing a backup.
+func (e *Engine) LockKVStore() {
+	e.mu.Lock()
```

Comment:
Locking the engine here might be overkill, but it was easy enough to do and might be prevent some edge issues